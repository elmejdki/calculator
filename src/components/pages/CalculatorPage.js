import React, { useReducer } from 'react';
import Display from '../Display';
import ButtonPanel from '../ButtonPanel';
import calculate from '../../logic/calculate';

const CalculatorPage = () => {
  const [state, dispatch] = useReducer((state, action) => {
    switch (action.type) {
      case 'CALCULATE':
        return calculate(state, action.operation);
      default:
        return state;
    }
  }, {
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = operation => {
    dispatch({ type: 'CALCULATE', operation });
  };

  const { next, total, operation } = state;

  return (
    <div className="flex flex-column app-container center">
      <Display
        result={
          `${(!total && next) || total || '0'}
           ${operation || ''}
           ${(total && next) || ''}`
        }
      />
      <ButtonPanel
        clickHandler={handleClick}
      />
    </div>
  );
};

export default CalculatorPage;
