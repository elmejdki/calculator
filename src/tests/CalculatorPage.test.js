import React from 'react';
import renderer from 'react-test-renderer';
import CalculatorPage from '../components/pages/CalculatorPage';

it('renders correctly the Calculator Page', () => {
  const tree = renderer.create(<CalculatorPage />).toJSON();

  expect(tree).toMatchSnapshot();
});
