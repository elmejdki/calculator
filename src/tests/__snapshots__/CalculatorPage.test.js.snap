// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the Calculator Page 1`] = `
<div
  className="flex flex-column app-container center"
>
  <p
    className="flex display"
    role="main"
  >
    0
           
           
  </p>
  <div
    className="flex flex-column"
  >
    <div>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button operation-btn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button operation-btn"
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button operation-btn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button operation-btn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button double-button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
